packages/server-side-graphql-client/README.md

```diff
@@ -0,0 +1,371 @@
+- it performs CRUD operations using the standard graphQL API which keystone provides
+- in addition to create, read, update and delete functions, it provides `runCustomQuery` to execute custom GraphQL via the same mechanism
+- by default it respects access-controls or can be run against the internal GraphQL schema to bypass access controls
+
+```js
+const { createItems } = require('@keystonejs/server-side-graphql-client');
+
+keystone.createList('User', {
+  fields: {
+    name: { type: Text },
+    email: { type: String },
+  },
+});
+
+const seedUser = async usersData => {
+  await createItems({ keystone, listName: 'User', items: usersData });
+};
+
+// seedUser would normally be called in a differnt script
+seedData([{ data: { name: 'keystone user', email: 'keystone@test.com' } }]);
+```
+
+## API
+
+To perform CRUD operations, you can use the following functions:
+
+- `createItem`
+- `createItems`
+- `updateItem`
+- `updateItems`
+- `deleteItem`
+- `deleteItems`
+- `runCustomQuery`
+
+> NOTE: All functions accepts an `config` object as an argument, and return a `Promise`.
+
+### Shared Config Options
+
+The following config options are common to all server-side graphQL functions.
+
+| Properties     | Type       | Default    | Description                                                                                                  |
+| -------------- | ---------- | ---------- | ------------------------------------------------------------------------------------------------------------ |
+| `keystone`     | `Keystone` | (required) | Keystone instance.                                                                                           |
+| `listName`     | `String`   | (required) | Keystone list name name.                                                                                     |
+| `returnFields` | `String`   | `id`       | A graphQL fragment of fields to return. Must match the graphQL return type.                                  |
+| `schemaName`   | `String`   | `public`   | Name of your GraphQL API schema. To override the access-control mechanism, provide `internal` as schemaName. |
+| `extraContext` | `Object`   | `{}`       | Additional context option object that gets passed onto `keystone.executeGraphQL` method.                     |
+
+### createItem
+
+Allows you to create single item.
+
+#### Usage
+
+```js
+const { createItem } = require('@keystonejs/server-side-graphql-client');
+
+keystone.createList('User', {
+  fields: {
+    name: { type: Text },
+    email: { type: String },
+  },
+});
+
+const addUser = async userInput => {
+  const user = await createItem({
+    keystone,
+    listName: 'User',
+    item: userInput,
+    returnFields: `name, email`,
+  });
+  console.log(user); // { name: 'keystone user', email: 'keystone@test.com'}
+};
+
+addUser({ name: 'keystone user', email: 'keystone@test.com' });
```

Comment:
Can we simplify this example (and the ones below) be removing the intermediate function `userInput`? Ideally we'd have the minimal example demonstrating the behaviour of the functions.

Reply:
I did this because the previous examples had:

```
// inside an asyc function
await ....
```

I agree that an asyc example is much neater to read than a promise, so I think it makes sense to show a tiny little function that demonstrates it is async as well as giving a bit of a hint about the types of functions users might want to create.

Reply:
Yeah, I think this is tricky. As is, these examples aren't cut and pastable, we're in a bit of a no-man's-land between cut-and-pastable. I think perhaps those kind of examples belong in a more learning oriented resource, like a data-seeding users guide, rather than the information-oriented API docs. We don't have to get this right straight away, let's just flag it as something to think about and loop back on in the future.

Reply:
Agree, as it is I think these are better structure than the previous examples -but we can probably do more in future